processing/src/main/java/org/apache/druid/query/InlineDataSource.java

```diff
@@ -125,7 +125,7 @@ private static boolean rowsEqual(final Iterable<Object[]> rowsA, final Iterable<
         final Object[] rowA = listA.get(i);
         final Object[] rowB = listB.get(i);
 
-        if (!Arrays.equals(rowA, rowB)) {
+        if (!Arrays.deepEquals(rowA, rowB)) {
```

Comment:
There's a `hashCode` method that claims that it is compatible with this method.  I'm not sure that is actually true anymore because I don't believe it walks into Arrays-of-Arrays the same way that this one does.